### PR TITLE
Remove secret key from git

### LINKED ISSUE #28: secret key storage
You should do two other things with respect to PR #13:

1.  make sure you use a new key that has never been checked into git
2. Put the secret_key file either outside of git ($HOME/.cardinal_key or something) or put the secret_key filename inside .gitignore

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**
fixes #28

### CODE REVIEW

**Comment by JakeRoggenbuck** [comment]
We do need a key for auto testing to work. Should I add a default key and specify it's not to be used in any place except github actions?

**Comment by hardaker** [comment]
If you can detect that you're in auto-tests (there should be a way right?), then you can just create a dummy one to use only when inside a test.

**Comment by JakeRoggenbuck** [comment]
Yea, we definitely could do that! I was trying to do that with 602bd but I didn't like that it would still work even when it was not being tested. Now I'm trying to see if github secret would work.

**Comment by JakeRoggenbuck** [comment]
It took a bit of trial and error but Its working as intended now!

**Comment by JakeRoggenbuck** [comment]
If it's run and the env variable DJANGO_SECRET_KEY is not set, it will warn about setting it in production. This does not error so it works for tests as well.

File: CardinalAPI/settings.py [hunk 1]
```diff
@@ -19,12 +20,13 @@
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/
 
-# SECURITY WARNING: keep the secret key used in production secret!
-with open("secret_key") as file:
-    SECRET_KEY = file.read().rstrip("\n")
+SECRET_KEY = os.environ.get("DJANGO_SECRET_KEY", False)
+if not SECRET_KEY:
```
Comment (hardaker) [inline CardinalAPI/settings.py:24]: So this isn't safe this way, as it only takes a human error to get it wrong.  If you can't detect you're inside the debugging environment itself, then can you simply error out here and instead put the secret_key environment variable in the testing environment itself?  Alternatively, you could programatically generate a new random key at run time if there isn't one set.

Comment (JakeRoggenbuck) [reply CardinalAPI/settings.py:24]: Yea, that's a good point. I'll try to have it use the env variable, error if it's not set. Then, in testing, set a env variable before tests are run.

Comment (hardaker) [reply CardinalAPI/settings.py:24]: perfect

**Comment by JakeRoggenbuck** [comment]
Okay, something to note is, there are two things I was confusing, `Environment secrets` and `Repository secrets`, I set the env secret but was trying to access the repo secret. I'll keep this in mind next time. The code will raise and exception when a key is not in the env and tests pass,

**Review by adamhutchings — APPROVED** [review]
Reviewed!

**Pull request merged**